### DISCUSSION TITLE
批量导出

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3494] [插件说明]
描述
---

> 批量导出任意进入阅读模式的页面。

地址
---

[https://simpread.ksria.cn/plugins/details/QLweMXPFWP](https://simpread.ksria.cn/plugins/details/QLweMXPFWP)

使用视频
---

https://user-images.githubusercontent.com/81074/157393140-b742aa25-35b4-4229-b431-cc6efe580b82.mp4

特点
---

1. 接入简悦的全部生产力工具

2. 因为阅读模式仅获取正文，相比其它的合并下载功能可以得到更加纯粹的「合辑」内容。

使用场景
---

- 可配合简悦的导出功能，将合并下载的内容导出为 Epub 结合 [利用 Vivaldi + 微信读书 + 文件传输助手 + 简悦让你脱离微信体系](https://zhuanlan.zhihu.com/p/466532702) 方案可方便的传到你的微信读书或 Kindle

- 配合 [使用白名单让任意页面实现自动进入阅读模式](https://github.com/Kenshin/simpread/discussions/1672)，可适配大多数小说类网页，如果你不了解此功能，可以在[此提交网站](https://github.com/Kenshin/simpread/discussions/3494#discussioncomment-2362864)，我帮你测试。

- 将批量打开的页面合并到当前页，方便在N合一之后的继续操作，如标注、加入稍后读等操作。

快捷键
---

> 支持全局快捷键 `b e`

选项
---

> 随便进入一个页面 e.g. https://sspai.com/post/71576 并进入 **阅读模式 → 右下角触发器 → 动作 → 插件触发器**

<img src="https://user-images.githubusercontent.com/81074/157398383-e304291f-023c-4a43-9e92-43d679ac728b.png" width="650" />

<img src="https://user-images.githubusercontent.com/81074/158335258-ec8eb3ad-e41a-4e6b-a648-936a86f44fb5.png" width="650" />

功能说明
---

1. 选择导出格式

    默认为 `HTML`，包括：[下载到本地](http://ksria.com/simpread/docs/#/%E4%BF%9D%E5%AD%98%E5%88%B0%E6%9C%AC%E5%9C%B0) 与 [导出到生产力工具](http://ksria.com/simpread/docs/#/%E5%AF%BC%E5%87%BA%E5%88%B0%E7%94%9F%E4%BA%A7%E5%8A%9B%E5%B7%A5%E5%85%B7)

   ![image](https://user-images.githubusercontent.com/81074/157390365-c1696eae-6574-4a2e-9a1f-d885ff368006.png)

2. 标题

    合并下载文件的标题，可为空，默认为 `合并下载-untitled`。

    ![image](https://user-images.githubusercontent.com/81074/157389364-f8ca36d4-e7eb-4e16-a984-ed9c85319759.png)

3. 批量打开的 URLs

    输入需要合并导出的 URL，每行一个，支持 `{n:m}` 方式，详细说明 https://github.com/Kenshin/simpread/discussions/3494#discussioncomment-10008357

4. 分别下载

    选中后，当打开这个 URL 后就会自动下载文件。

    ![image](https://user-images.githubusercontent.com/81074/157390070-0c347040-dc54-4d61-b133-c6f924d68244.png)

5. 是否包含当前页面

    默认为不包含，选中后批量导出的内容会包含当前页面，并且位置在第一个。

    ![image](https://user-images.githubusercontent.com/81074/157391209-bea02ab1-6440-45ae-bb05-493ad55cb41c.png)

6. 全部打开与合并下载

    这两个功能是配合使用的，首先使用 **全部打开** 根据 URLs 规则打开，然后手动点击 **合并下载** 将打开的 URLs 合并到一个文件，并下载到本地。

   ![image](https://user-images.githubusercontent.com/81074/157390728-014ff072-fdc5-4fc1-aa74-ccd824342aba.png)

7. 合并到当前页

    将批量打开的页面合并到当前页，方便在N合一之后的继续操作，如标注、加入稍后读等操作。

注意事项
---

详细说明 [请看这里](https://github.com/Kenshin/simpread/discussions/3494#discussioncomment-2321724)。

更新日志
---

### 1.0.0

- 快捷键 `b e`

- 输入框
    - 根据输入框中的 URLs 批量打开

    - 根据特定规则打开 URLs

- 标记每个打开的 URL 并保存到 storage 中

- 目录

- 合并下载时的名称

- 合并下载
    将打开的 URLs 下载为同一个文件

- 分别下载

- 合并到当前页

- 选择导出方案

- 是否包含当前页面

引申
---

> 安装插件或带有插件触发器的插件失效的 [解决方案](https://github.com/Kenshin/simpread/discussions/2342)。

**Comment by Kenshin** [comment]
注意事项
---

1. 可配合 [增强导出](https://kb.simpread.pro/#/page/增强导出)，将合并下载的内容导出到任意文件夹。
2. 批量打开的页面需要进入阅读模式后才能使用此功能，所以最好的方案是当批量打开后，手动确认是否已进入阅读模式。
3. 当你使用 [知识库](https://kb.simpread.pro/#/page/建立知识库) 功能的话，想使用批量导出需要调整你的配置方案，需要调整下面的选项。

> 选项页 → 服务 → 定制导出，关闭下面的选项

<img src="https://user-images.githubusercontent.com/81074/157378349-e49d6ca0-99fe-4076-bde5-a89b0ef35245.png" width="600" />

> 选项页 → 高级设定 → 标注，确保跟下面一致

<img src="https://user-images.githubusercontent.com/81074/157378388-dd41f940-041d-4b0b-b985-ac6d94eece10.png" width="600" />

**Reply by byheaven** [reply]
感觉这个和知识库的冲突比较影响使用体验，能不能改成在使用批量导出插件的时候，临时覆盖这些配置参数，批量导出完成后自动复原？

**Reply by Kenshin** [reply]
> 感觉这个和知识库的冲突比较影响使用体验，能不能改成在使用批量导出插件的时候，临时覆盖这些配置参数，批量导出完成后自动复原？

确实如此，当前简悦的版本暂时没办法实现，因为安全性因素，所以简悦插件无法获取并写入用户数据（配置文件）

只能下个版本解决。

PS：好在这个方式并不会经常使用。 😂

**Reply by DanShui-Yu** [reply]
请问现在解决了吗？

**Reply by Kenshin** [reply]
@PupupupuTong 

没有，如果解决了我会在这里通知的~

**Reply by sunyou-19** [reply]
可不可以在稍后读增加这个批量导出的功能。

**Reply by Kenshin** [reply]
@sunyou-19 

> 可不可以在稍后读增加这个批量导出的功能。

可以考虑。

***

另外，没有插件触发器的解决方案 https://www.yuque.com/kenshin/simpread/fcw60u

PS：不要在无关的地方提问，如有需要请通过 https://github.com/kenshin/simpread/issues/new 讨论。

**Comment by Kenshin** [comment]
小说类网站汇总
---

利用 [使用白名单让任意页面实现自动进入阅读模式](https://github.com/Kenshin/simpread/discussions/1672) 可适配大多数小说类网页自动进入阅读模式。

提交
---

如果你不了解此功能，可以在此提交网站，我帮你测试。

**Comment by bighaidao** [comment]
请问是否可以增加 不合并下载的选项。 目前有个需求想批量保存，但是并不想合并。 因为内容太大，合并在一起非常头疼

**Reply by Kenshin** [reply]
emmm 考虑下这个需求。

**Comment by Kenshin** [comment]
# 批量打开 URLs 时支持规则

1. 普通 URL e.g. https://sspai.com/post/71576

2. 批量模式，使用占位符号 `{n:m}`

    - `n`: 起始数字
    - `m`: 结束数字

     e.g. `https://sspai.com/prime/story/specialty-coffee-part-{1:3}` 会自动打开下面三个 URL

     - [https://sspai.com/prime/story/specialty-coffee-part-1](https://sspai.com/prime/story/specialty-coffee-part-1)
     - [https://sspai.com/prime/story/specialty-coffee-part-2](https://sspai.com/prime/story/specialty-coffee-part-2)
     - [https://sspai.com/prime/story/specialty-coffee-part-3](https://sspai.com/prime/story/specialty-coffee-part-3)
     
#### 一个例子

将下面的代码粘贴到输入框

```
https://sspai.com/post/71576
https://sspai.com/post/69972
https://sspai.com/post/70237
```

**Comment by Kenshin** [comment]
# 占位

**Comment by Kenshin** [comment]
# 占位

**Comment by Kenshin** [comment]
# 占位
